### ISSUE TITLE
About ImageDataGenerator

### ISSUE THREAD

**Issue by lokinfey** [SciSharp/Keras.NET#16]
When I use

                var train_dataset = train_data.FlowFromDirectory(directory : "home/lokinfey/ml/ImageRecog/dataset/train" , target_size : (64,64).ToTuple(),batch_size : 32, class_mode :"binary");

it show me this error :

Exception has occurred: CLR/Python.Runtime.PythonException
An unhandled exception of type 'Python.Runtime.PythonException' occurred in Python.Runtime.NETStandard.dll
at Python.Runtime.PyObject.Invoke(PyTuple args, PyDict kw)
at Python.Runtime.PyObject.InvokeMethod(String name, PyTuple args, PyDict kw)
at Keras.Base.InvokeMethod(String method, Dictionary2 args) in /home/lokinfey/dev/Keras.NET/Keras/Base.cs:line 116 at Keras.PreProcessing.Image.ImageDataGenerator.FlowFromDirectory(String directory, Tuple2 target_size, String color_mode, String[] classes, String class_mode, Int32 batch_size, Boolean shuffle, Nullable`1 seed, String save_to_dir, String save_prefix, String save_format, Boolean follow_links, String subset, String interpolation) in /home/lokinfey/dev/Keras.NET/Keras/PreProcessing/ImageDataGenerator.cs:line 219
at ImageRecog.Program.Main(String[] args) in /home/lokinfey/ml/ImageRecog/Program.cs:line 39

**Comment by deepakkumar1984** [comment]
Fixed in the code. Please use the latest code to test your scenario. I will update nuget later with other fixes.

**Comment by lokinfey** [comment]
But when I use Model.FitGenerator  it show me that Argument 1: cannot convert from 'Keras.KerasIterator' to 'Keras.Utils.Sequence' (CS1503) [KerasDemo]

**Comment by deepakkumar1984** [comment]
Ok I will have a look

**Comment by deepakkumar1984** [comment]
Can you share me sample code, will be easy for me to debug quickly?

**Comment by lokinfey** [comment]
var model = new Sequential();
                model.Add(new Conv2D(32,(3,3).ToTuple(), input_shape: new Shape(64,64,3), activation:"relu"));

                model.Add(new MaxPooling2D( pool_size: (2,2).ToTuple() ));

                model.Add(new Flatten());

                model.Add(new Dense(128, activation:"relu"));
                
                model.Add(new Dense(1, activation: "sigmoid"));
                
                model.Compile( optimizer :"adam" ,  loss: "binary_crossentropy" , metrics :  new string[]{"accuracy"} );


                var datagen = new  ImageDataGenerator( rescale : (float)1.0/255 ,  shear_range : (float)0.2 , zoom_range : (float)0.2 , horizontal_flip : true  );

                //var train_datta = datagen.FlowFromDirectory("dataset/", class_mode:"binary", batch_size:64);

                var train_dataset = datagen.FlowFromDirectory( directory: "dataset/training_set" , target_size : (64,64).ToTuple(),batch_size : 32 , class_mode :"binary");

                
                var test_dataset = datagen.FlowFromDirectory( directory: "dataset/test_set" , target_size : (64,64).ToTuple(),batch_size : 32 , class_mode :"binary");

                model.FitGenerator(train_dataset , steps_per_epoch : 8000 , epochs :25 , validation_steps : 800 , validation_data :  test_dataset ) ;

//model.FitGenerator error cannot convert from 'Keras.KerasIterator' to 'Keras.Utils.Sequence' (CS1503) [KerasDemo]

This is Keras document about that , https://keras.io/preprocessing/image/ 

train_datagen = ImageDataGenerator(
        rescale=1./255,
        shear_range=0.2,
        zoom_range=0.2,
        horizontal_flip=True)

test_datagen = ImageDataGenerator(rescale=1./255)

train_generator = train_datagen.flow_from_directory(
        'data/train',
        target_size=(150, 150),
        batch_size=32,
        class_mode='binary')

validation_generator = test_datagen.flow_from_directory(
        'data/validation',
        target_size=(150, 150),
        batch_size=32,
        class_mode='binary')

model.fit_generator(
        train_generator,
        steps_per_epoch=2000,
        epochs=50,
        validation_data=validation_generator,
        validation_steps=800)

**Comment by deepakkumar1984** [comment]
It's fixed now. I have tested your code with cats and dogs dataset and I am able to run the training. Please pull the latest code and let me know how it goes

**Comment by lokinfey** [comment]
thank you ! It is cool now
